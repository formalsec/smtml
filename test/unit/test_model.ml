open Smtml

(** Test Model.to_json *)
let () =
  let x = Symbol.make Ty_int "x" in
  let y = Symbol.make Ty_real "y" in
  let z = Symbol.make Ty_bool "z" in
  let u = Symbol.make Ty_str "u" in
  let model : Model.t =
    let tbl = Hashtbl.create 16 in
    List.iter
      (fun ((s, v) : Symbol.t * Value.t) -> Hashtbl.replace tbl s v)
      [ (x, Int 1); (y, Real 2.0); (z, True); (u, Str "abc") ];
    tbl
  in
  let model_to_json = Model.to_json model in
  Format.printf "%a@." (Yojson.pretty_print ~std:true) model_to_json
