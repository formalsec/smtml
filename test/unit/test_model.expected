{
  "model": [
    { "name": "u", "ty": "str", "value": "abc" },
    { "name": "z", "ty": "bool", "value": true },
    { "name": "x", "ty": "int", "value": 1 },
    { "name": "y", "ty": "real", "value": 2.0 }
  ]
}
